cypress_test/integration_tests/desktop/calc/top_toolbar_spec.js

```diff
@@ -17,17 +17,61 @@ describe('Top toolbar tests.', function() {
 		helper.afterAll(testFileName, this.currentTest.state);
 	});
 
+	function getTextEndPosForFirstCell() {
+		calcHelper.dblClickOnFirstCell();
+
+		helper.getCursorPos('left', 'currentTextEndPos');
+	}
 
 	it('Merge cells', function() {
-		calcHelper.selectEntireSheet();
 
-		cy.get('#tb_editbar_item_togglemergecells')
+		// Select the full row
+		calcHelper.selectFirstColumn();
+	
+		// Despite the selection is there, merge cells needs more time here.
+		cy.wait(1000);
+	
+		cy.get('.w2ui-tb-image.w2ui-icon.togglemergecells')
 			.click();
-		
-		cy.get('.lokdialog_canvas')
+	
+		cy.get('.ui-dialog.ui-corner-all.ui-widget.ui-widget-content.ui-front.lokdialog_container')
 			.should('exist');
```

Comment:
There is a helper method for dialogs: checkDialogAndClose() in helper.js. It needs the dialog title as a parameter, so it can check whether the right dialog is opened. Can you use that here?